src/Microsoft.ServiceFabric.Services/Communication/Client/OperationRetrySettings.cs

```diff
@@ -65,7 +98,16 @@ public TimeSpan MaxRetryBackoffIntervalOnTransientErrors
         /// <value>Maximum retry interval to back-off on non transient errors</value>
         public TimeSpan MaxRetryBackoffIntervalOnNonTransientErrors
         {
-            get { return this.maxRetryBackoffIntervalOnNonTransientErrors; }
+            get
+            {
+                var contantpolicy = this.retryPolicy as ConstantRetryPolicy;
+                if (contantpolicy != null)
+                {
+                    return contantpolicy.MaxRetryBackoffIntervalOnNonTransientErrors;
+                }
+
+                throw new NotSupportedException("This retry Policy doesn't support this functionality ");
+            }

```

Comment:
This casting of the interface looks weird. You can put a note saying this is done just for backward compat.